woocommerce/woocommerce-android
# Add see report button in Analytic Hub

Closes: #10666
Closes: #10667

### Why
As part of the Expanded Analytic Hub project, we decided to support opening fully authenticated WebViews to display the full report for each card in the Analytics Hub.

### Description
This PR adds support for displaying the full report for each card in the Analytics Hub by adding a `See Report` section to the two types of cards we have on the Analytics Hub. When this new See Report section is pressed, it will trigger an event to open a WebView with the report URL. For WP.com stores, the Webview where we open the report will contain the authentication information, so merchants should be able to see the report without the need to re-authenticate in the browser. For stores that are not WP.com stores, we will open the default WebView, so if merchants are not logged into the device's browser, they will need to log in to the web to see the report.

### Testing instructions
TC1
1. Open the App
2. Tap on See More to enter the Analytic Hub screen.
3. Check that Orders, Products, and Revenue have the new See Report section
4. Check that Session doesn't have the new  See Report section
5. Change the selected granularity and tap on See report
6. Check that the granularity selected on the device matches the one selected on the web
7. Additionally, select the custom option and select a date range
8. Tap on  the See Report section
9. Check that the same date range is also selected on the web

TC2 ( For WP.com stores)
1. Check that your account is logged out from the web
2. Open the app 
3. Tap on See More to enter the Analytic Hub screen.
4. Tap on the See Report section and check that the correct report is displayed on wp-admin.

TC3 ( For non WP.com stores)
1. Check that your account is logged out from the web
2. Open the app 
3. Tap on See More to enter the Analytic Hub screen.
4. Tap on the See Report section and check that the web asks you to log in
6. Once logged in, check that the correct report is displayed on wp-admin.
7. Go back to the app
8. Open a different report
9. Check that this time, the report is opened without needing to authenticate in the web



### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/e270cab9-0e3f-498b-945a-3421527ddeb8
